packages/common-utils/src/types.ts

```diff
@@ -115,6 +115,23 @@ export const SelectSQLStatementSchema = z.object({
   havingLanguage: SearchConditionLanguageSchema.optional(),
   orderBy: SortSpecificationListSchema.optional(),
   limit: LimitSchema.optional(),
+  with: z
+    .array(
+      z.object({
+        name: z.string(),
+        sql: z.object({
+          sql: z.string(),
+          params: z.record(z.string(), z.any()),
+        }),
+        // If true, it'll render as WITH ident AS (subquery)
+        // If false, it'll be a "variable" ex. WITH (sql) AS ident
+        // where sql can be any expression, ex. a constant string
+        // see: https://clickhouse.com/docs/sql-reference/statements/select/with#syntax
+        // default assume true
+        isSubquery: z.boolean().optional(),
```

Comment:
style: maybe we can call it `isVariable`

Reply:
i'm not sure what that means? subquery seems to be the term used in the CH syntax docs

Reply:
nvm, I was thinking of the other types of CTE. I think the subquery is a cleaner flag for now